server/models/User.js

```diff
@@ -477,60 +484,52 @@ export default (Sequelize, DataTypes) => {
-          collectiveName = 'anonymous';
-        } else if (slugify(collectiveName).length === 0) {
-          collectiveName = 'user';
-        }
-
-        const userCollective = {
-          type: 'USER',
-          name: collectiveName,
-          image: userData.image,
-          mission: userData.mission,
-          description: userData.description,
-          longDescription: userData.longDescription,
-          website: userData.website,
-          twitterHandle: userData.twitterHandle,
-          githubHandle: userData.githubHandle,
-          currency: userData.currency,
-          hostFeePercent: userData.hostFeePercent,
-          isActive: true,
-          CreatedByUserId: userData.CreatedByUserId || user.id,
-          data: { UserId: user.id },
-        };
-        return models.Collective.create(userCollective, sequelizeParams);
-      })
-      .tap(collective => {
-        // It's difficult to predict when the image will be updated by findImageForUser
-        // So we skip that in test environment to make it more predictable
-        if (config.env !== 'test' && config.env !== 'circleci') {
-          collective.findImageForUser(user);
-        }
-        user.CollectiveId = collective.id;
-        return user.save(sequelizeParams);
-      })
-      .then(collective => {
-        user.collective = collective;
-        return user;
-      });
+    const user = await User.create(userData, sequelizeParams);
+    let name = userData.firstName;
+    if (name && userData.lastName) {
+      name += ` ${userData.lastName}`;
+    }
+
+    // If user doesn't provide a name, set it to "anonymous". If we cannot
+    // slugify it (for example firstName="------") then fallback on "user".
+    let collectiveName = userData.name || name;
+    if (!collectiveName || collectiveName.trim().length === 0) {
+      collectiveName = 'anonymous';
+    } else if (slugify(collectiveName).length === 0) {
+      collectiveName = 'user';
+    }
+
+    const userCollectiveData = {
+      type: 'USER',
+      name: collectiveName,
+      image: userData.image,
+      mission: userData.mission,
+      description: userData.description,
+      longDescription: userData.longDescription,
+      website: userData.website,
+      twitterHandle: userData.twitterHandle,
+      githubHandle: userData.githubHandle,
+      currency: userData.currency,
+      hostFeePercent: userData.hostFeePercent,
+      isActive: true,
+      CreatedByUserId: userData.CreatedByUserId || user.id,
+      data: { UserId: user.id },
+    };
+    user.collective = await models.Collective.create(userCollectiveData, sequelizeParams);
+
+    // It's difficult to predict when the image will be updated by findImageForUser
+    // So we skip that in test environment to make it more predictable
+    if (config.env !== 'test' && config.env !== 'circleci') {
+      user.collective.findImageForUser(user);
+    }
+    user.CollectiveId = user.collective.id;
+    await user.save(sequelizeParams);
```

Comment:
I'd rather return the result of `user.save(sequelizeParams)` directly, as formatters apply to the data when saving the model before and after `save` can be different.

Reply:
it doesn't matter since we are waiting for the result of `user.save`.
But sure. Doesn't hurt.

Reply:
Does `user.save` mutates the object with the return?

Reply:
I'd expect so. But I'm not sure. Doesn't matter here.

Reply:
This was a genuine question, I don't know if `save` mutates the object and sequelize's documentation isn't helping much.